From 800f336b0bfbe6bc932d438eac5ccce6ab7196e3 Mon Sep 17 00:00:00 2001
From: Hannes Duerr <h.duerr@proxmox.com>
Date: Wed, 16 Jul 2025 10:01:27 +0200
Subject: [PATCH] tools: fix daemon starting order for debian packages

Currently the management daemon (mgmtd) starts only after Zebra is
started, which causes Zebra to throw some errors as described here [0].
The problem is, when the MGMTD was introduced [1], the order in tools/frr.in
was adjusted correctly, but not in tools/frrcommon.sh.in.
We are therefore now also changing it in frrcommon.sh.in.

[0] https://github.com/FRRouting/frr/issues/17931
[1] https://github.com/FRRouting/frr/pull/13059/commits/1c84efe4fa8585df58a9b53459f94c47934f0786

Signed-off-by: Hannes Duerr <h.duerr@proxmox.com>
---
 tools/frrcommon.sh.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/frrcommon.sh.in b/tools/frrcommon.sh.in
index 44355f8895..1848841aff 100755
--- a/tools/frrcommon.sh.in
+++ b/tools/frrcommon.sh.in
@@ -36,7 +36,7 @@ FRR_DEFAULT_PROFILE="@DFLT_NAME@" # traditional / datacenter
 # - keep zebra first
 # - watchfrr does NOT belong in this list
 
-DAEMONS="zebra mgmtd bgpd ripd ripngd ospfd ospf6d isisd babeld pimd pim6d ldpd nhrpd eigrpd sharpd pbrd staticd bfdd fabricd vrrpd pathd"
+DAEMONS="mgmtd zebra bgpd ripd ripngd ospfd ospf6d isisd babeld pimd pim6d ldpd nhrpd eigrpd sharpd pbrd staticd bfdd fabricd vrrpd pathd"
 RELOAD_SCRIPT="$D_PATH/frr-reload.py"
 
 #
-- 
2.47.2

